Apps/Sandcastle/CesiumSandcastle.js

```diff
@@ -810,6 +811,14 @@ require({
         }
     }
 
+    registry.byId('buttonNew').on('click', function() {
+        loadFromGallery(newDemo);
+        var demoSrc = newDemo.name + '.html';
+        if (demoSrc !== window.location.search.substring(1)) {
```

Comment:
You're right. Sorry I didn't notice that earlier. It's because location.search just spits out the entire query URL and so the string matching fails. I would need to split the string into separate query strings and then get the value for the 'src' key. That should fix it.